### DISCUSSION TITLE
Is MediaMTX WebRTC really able to work with trickle ice ?

### DISCUSSION THREAD

**Discussion by neilyoung** [bluenviron/mediamtx#3194] [Questions]
### Question

I have some doubts. I first tested a client WHIP implementation in VueJS. When Trickle-ICE was enabled, MediaMTX panicked.
It worked with ICE LITE.

Now I'm having a second solution. There is no panic now, but also no success. I see PATCH requests leaving, and these are the important ones (srflx) in this network scenario...

In the end the publish doesn't work, because ICE does not work...

**Comment by neilyoung** [comment]
No it doesn't support trickle ICE at least not via the WHEP endpoint. It is also not properly reacting to PATCH request transferring new ICE credentials.

**Reply by aler9** [reply]
this is false, trickle ICE is fully supported and asynchronous ICE candidates can be sent through PATCH requests. If you are having issues, open a bug report and provide clear details on how to replicate, with logs and everything else.

**Reply by neilyoung** [reply]
OK, thanks for clarification. May I ask for another clarification? Your WHEP implemtation, as you say, supports trickle ICE. I have seen, that the PATCH requests are answered 204, so seemingly correct. But wouldn't a WHEP implementation have to immediately after this having to start an ICE BINDING Request for the new candidate? And wouldn't it have to immediately take over the provided ice:ufrag, ice:pwd? The latter is what I'm missing.

**Reply by neilyoung** [reply]
But maybe even this is ok. You are responding 204 no content, hence indicating, that you are not doing an ICE restart. 

"A WHEP resource receiving a PATCH request with new ICE candidates, but which does not perform an ICE restart, MUST return a "204 No Content" response without body. If the Media Server does not support a candidate transport or is not able to resolve the connection address, it MUST accept the HTTP request with the "204 No Content" response and silently discard the candidate."

Could you please confirm this to be the current MTX behaviour? That would explain something I see.

**Reply by aler9** [reply]
I'm forced to close this discussion since required details were not provided properly. Please follow the rules before asking for help here.
